File: archive/backups/v1.2.0_2026-01-07_13-02/src/lib/types/weapons.ts
export interface Weapon {
  id: string;
  name: string;
  category: string;
  weapon_type: string;
  damage_dice: string;
  damage_type: string;
  weight_kg: number;
  cost_gp: number;
  data: WeaponData;
  source: 'core' | 'homebrew' | 'override';
}

export interface WeaponData {
  properties: string[];
  mastery: string;
  range?: WeaponRange;
  thrown_range?: WeaponRange;
  versatile_damage?: string;
  ammunition_type?: string;
  property_details?: Record<string, PropertyDetail>;
  mastery_details?: MasteryDetail;
  source_page: number;
}

export interface WeaponRange {
  normal: number;
  max: number;
}

export interface PropertyDetail {
  id: string;
  name: string;
  description: string;
  data: any;
}

export interface MasteryDetail {
  id: string;
  name: string;
  description: string;
  data: any;
}

File: archive/backups/v1.2.0_2026-01-07_13-02/src-tauri/src/types/weapons.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Weapon {
    pub id: String,
    pub name: String,
    pub category: String,
    pub weapon_type: String,
    pub damage_dice: String,
    pub damage_type: String,
    pub weight_kg: f64,
    pub cost_gp: f64,
    pub data: WeaponData,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct WeaponData {
    pub properties: Vec<String>,
    pub mastery: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub range: Option<WeaponRange>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub thrown_range: Option<WeaponRange>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub versatile_damage: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub ammunition_type: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub property_details: Option<HashMap<String, PropertyDetail>>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub mastery_details: Option<MasteryDetail>,
    pub source_page: u16,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct WeaponRange {
    pub normal: f64,
    pub max: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PropertyDetail {
    pub id: String,
    pub name: String,
    pub description: String,
    pub data: serde_json::Value,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MasteryDetail {
    pub id: String,
    pub name: String,
    pub description: String,
    pub data: serde_json::Value,
}






